pkg/daemon/daemon_test.go

```diff
@@ -0,0 +1,265 @@
+package daemon
+
+import (
+	"context"
+	"flag"
+	"testing"
+
+	sriovnetworkv1 "github.com/k8snetworkplumbingwg/sriov-network-operator/api/v1"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/fakefilesystem"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	snclientset "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/client/clientset/versioned"
+	fakesnclientset "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/client/clientset/versioned/fake"
+	plugin "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/plugins"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/plugins/generic"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/utils"
+
+	fakemcclientset "github.com/openshift/machine-config-operator/pkg/generated/clientset/versioned/fake"
+
+	fakek8s "k8s.io/client-go/kubernetes/fake"
+)
+
+var FakeSupportedNicIDs corev1.ConfigMap = corev1.ConfigMap{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      sriovnetworkv1.SupportedNicIDConfigmap,
+		Namespace: namespace,
+	},
+	Data: map[string]string{},
+}
+
+var SriovDevicePluginPod corev1.Pod = corev1.Pod{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      "sriov-device-plugin-xxxx",
+		Namespace: namespace,
+		Labels: map[string]string{
+			"app": "sriov-device-plugin",
+		},
+	},
+	Spec: corev1.PodSpec{
+		NodeName: "test-node",
+	},
+}
+
+func init() {
+	// Increase verbosity to help debugging failures
+	flag.Set("logtostderr", "true")
+	flag.Set("stderrthreshold", "WARNING")
+	flag.Set("v", "2")
+}
+
+var _ = Describe("Config Daemon", func() {
+	var stopCh chan struct{}
+	var syncCh chan struct{}
+	var exitCh chan error
+	var refreshCh chan Message
+
+	var cleanFakeFs func()
+
+	var sut *Daemon
+
+	BeforeEach(func() {
+		stopCh = make(chan struct{})
+		refreshCh = make(chan Message, 1024)
+		exitCh = make(chan error)
+		syncCh = make(chan struct{}, 64)
```

Comment:
why we need to fill this channel will 64 messages?

Reply:
It simulates `NodeStateStatusWriter` writing on the channel, it avoids the daemon waiting.